### ISSUE TITLE
[Internship Project]: Simran Tandon

### ISSUE THREAD

**Issue by GemmaTuron** [ersilia-os/ersilia#716]
### Summary

Hello,

This is a public issue for a virtual daily stand-up. We will use this to briefly share the tasks of the day and the challenges and advances made, so that we can ensure smooth support from the Ersilia mentors and alignment between daily tasks and overall internship goals.

### Scope

Initiative 🐋

### Objective(s)

Internship goals:

- Learn how to work in an Open Source Community
- Improve the infrastructure of the Ersilia Model Hub
- Improve the usability and platform compatibility of Ersilia Models, mainly via Docker Images
- Learn about AI/ML models for drug discovery
- Identify new relevant AI/ML models and datasets and add them to this [discussion](https://github.com/ersilia-os/ersilia/discussions/694)
- Incorporate new AI/ML models in the Hub
- Improve the documentation of the Ersilia Model Hub for users
- Improve the documentation of the Ersilia Model Hub for developers

### Team

| Role & Responsibility | Username(s) | 
|--|--|
| Intern | @simrantan |
| Mentor| @miquelduranfrigola  |
| Coordinator| @GemmaTuron  |

### Timeline

Before starting your work, line up a few tasks and short description. This should not take long. For example, it could be something like:
Wednesday 21st June
- [ ] Create Intern daily task tracker
- [ ] Update the Ersilia GitHub Project
- [ ] Make sure Git Actions start upon success of the other workflows
- [ ] Work on the documentation for model testing
- [ ] Attend the Drug Discovery intro by WCAIR

**Comment by GemmaTuron** [comment]
In this example, following the tasks of Wednesday 21st June:

- [x]  Create Intern daily task tracker
- [x]  Update the Ersilia GitHub Project
- [x]  Make sure Git Actions start upon success of the other workflows
- [ ]  Work on the documentation for model testing
- [x]  Attend the Drug Discovery intro by WCAIR

I have created all the templates for the Interns, spend a couple of hours revising the [GitHub Project](https://github.com/orgs/ersilia-os/projects/1) and updating the tasks. I have been working on identifying the bug in the GitActions, solved in this [issue](https://github.com/ersilia-os/eos-template/issues/29).
I have set a meeting with @miquelduranfrigola to discuss in detail his comments on the [Model testing discussion](https://github.com/ersilia-os/ersilia/discussions/700) but I haven't been able to start writing yet, i am to do so by the end of the week.
Looking forward to the WCAIR next lesson!

Pro tip: adding the links to the issues and discussions you mention will be very helpful!

**Comment by simrantan** [comment]
**6/23**

- [ ] Tested eos92sw on Docker, CLI and Colab
- [ ] Download and debug eos2hbd
- [ ] run --repo_path on eos2hbd 
- [ ] Test eos2v11 on Docker, CLI and Colab

**Comment by GemmaTuron** [comment]
Hi @simrantan !

Thansk for the summary, I've added a few pointers to help with the issoes on eos2v11

**Comment by simrantan** [comment]
Reflection 6/26

- [ ] Test eos22io on Docker, CLI, Colab
- [ ] Re-test eos2v11 on Docker, CLI, Colab 
- [ ] Debug eos2hbd
- [ ] Refactor eos2hbd


I tested eos22io on Docker, CLI, and Colab, finding no issues when I ran the [model](https://github.com/ersilia-os/eos22io/issues/7#issuecomment-1608578187). I also finally solved the eos2hbd error and was able to fetch the model and test it. I refactored according to instructions and plan to test the changed version locally tomorrow and submit a PR if it does.

**Comment by simrantan** [comment]
Reflection 6/27

- [ ] Test eos24jm on Docker, CLI, Colab 
- [ ] Finish refactoring eos2hbd
- [ ] Test eos2hbd locally
- [ ] submit PR for eos2hbd

I was able to test eos24jm with no issues on all [platforms](https://github.com/ersilia-os/eos24jm/issues/3#issuecomment-1610269707). I also finished refactoring eos2hbd. The first refactor I did yesterday wasn't working when I tested locally, but after some bug hunting I found that I had created a syntax error when editing service.py that was preventing the model from being fetched successfully. I fixed the error, re-cloned and tested the model, and it worked on the inputs I[ tried! ](https://github.com/ersilia-os/eos2hbd/pull/6#issue-1777830902). 
Besides testing eos2v11 when it is ready again, I am open for more new tasks!

**Comment by GemmaTuron** [comment]
I've added a couple of models to work on and some model testing!

**Comment by simrantan** [comment]
- [ ] Test eos46vmev
- [ ] review suggested models to work on
- [ ] begin working on one of the suggested models
I tried to open an issue with one of the drafts suggestions I saw in the tasks by subprojects chart but I think I messed up by setting the repo to ersilia instead of the model's[ repo](https://github.com/ersilia-os/ersilia/issues/727) - I am working on fixing it, so sorry about that! Also, for future reference - if issues are drafts should I wait for them to be opened by you or Miquel or should I open it myself(properly, not like what I did this time haha)

**Comment by GemmaTuron** [comment]
Hi @simrantan 

Apologies, I should have converted the draft to an issue, it is fixed now.

**Comment by simrantan** [comment]
- [ ] Refactor eos3zur 
- [ ] Test eos4zfy
- [ ] Test eos6aun
- [ ] Investigate error in workflow for eos1amn

**Comment by GemmaTuron** [comment]
@simrantan apologies for the eos1amn
I've assigned you new models,a nd alsor equesting your assistance to work with Riley and Febie since you have a Ubuntu installation

**Comment by simrantan** [comment]
Awesome! Happy to help. 
6/30 Goals/Reflection
- [ ] Help Riley (details: try fetching eos96ia on my device)
- [ ] Help Febie ()
- [ ] Refactor eos4se9
- [ ] Investigate why Dockerfile is taking so long to upload for eos2rd8/wait to see if it will upload 
- [ ] Refactor eos2rd8

**Comment by simrantan** [comment]
- [ ] Help Febie test eos1af5 on Windows
- [ ] Refactor eos4se9
- [ ] test eos4se9 locally
- [ ] Test eos6pbf on CLI, Colab, and Docker 
- [ ] Test eos3cf4 on CLI, Colab, and Docker
- [ ] Refactor eos2rd8*
- [ ] Test eos2rd8 locally*
*if time permits

I worked with Febie to test eos1af5 on Windows and ended up finding a new possible source of issues in the model (some package or version issues?) through some attempts at fetching and running the model locally. I also refactored eos4se9, and tested it locally, finding a bug that I am working on [resolving. ](https://github.com/ersilia-os/eos4se9/issues/3#issuecomment-1619255119)I also tested eos3cf4 on CLI and Colab, finding the same error as Hellen (module error) on both platforms.

**Comment by GemmaTuron** [comment]
Hi @simrantan 
Thanks for the detailed explanations, I have left some comments on model `eos1af5`. @emmakodes will have a look at `eos3cf4`. I don't see the PR for `eos4se9`. Once all this and model testing is complete I'll assign you new models!

**Comment by simrantan** [comment]
- [ ] Lab Meeting (8-9)
- [ ] Internship Meeting (10-10:30)
- [ ] Test eos6pbf on Docker 
- [ ] Test eos2rd8 locally 
- [ ] Check how long eos4se9 takes on 20 smiles 
- [ ] Submit PR for eos4se9
- [ ] Refactor eos3zur
- [ ] Test eos3zur locally

**Comment by GemmaTuron** [comment]
Hi @simrantan 
I've merged the PRs and given some tips for model eos4se9.
I've assigned a new model as well in case you finish and want to move on

**Comment by simrantan** [comment]
- [ ] Clean up Conda environments and models
- [ ] Continue testing eos4se9
- [ ] Investigate how to upload eos2rd8 to dockerhub (no space error)
- [ ] test eos6pbf on Docker

**Comment by GemmaTuron** [comment]
Perfect,

I have assigned model testing for `eos4se9` and merged the PR on `eos8h6g`.
Please try to see if there is something we could do on `eos2rd8`-- @samuelmaina did a good job on `eos7pw8` by caching some files and saving memory
I've assigned you a new model as well in case you finish with the rest of tasks

**Comment by simrantan** [comment]
Thank you for the tip! 

- [ ] Study eos7pw8
- [ ] Implement caching for some files in eos2rd8
- [ ] Test eos7jio
- [ ] Refactor eos5axz
- [ ] Re-test eos3cf4

Today, I spent a good chunk of time looking at eos7pw8 back in time and the current version with Samuel's additions to see how he implemented the caching in the model. I also looked at some online resources to better understand caching in python, and began identifying files that could be cached in eos2rd8. Tomorrow, I will implement the caching and test the model to ensure it still works. Is there a way to know the size of a model before and after caching a file? Or is it manual subtraction of the file size from the overall size?
I also tested [eos7jio](https://github.com/ersilia-os/eos7jio/issues/4#issuecomment-1624776395) on CLI, Colab and Docker. I have tested eos5axz locally to make sure it works, and once I confirmed it did I refactored the model and will test again tomorrow to ensure it is still functional. I also retested [eos3cf4 ](https://github.com/ersilia-os/eos3cf4/issues/3#issuecomment-1624794708). It is failing to upload to Colab still.

**Comment by GemmaTuron** [comment]
Thanks for the explanations @simrantan !
Let's focus on the two open models, eos2rd8 and eos5axz. I'd suggest using the templates created by @emmakodes to test if the caching reduces total size!

**Comment by simrantan** [comment]
- [ ] Test eos5axz locally
- [ ] Submit PR for eos5axz
- [ ] Implement caching for eos2rd8
- [ ] Test eos2rd8

Committed my refactored version and submitted PR for [eos5axz](https://github.com/ersilia-os/eos5axz/pull/7#issue-1794377167)

**Comment by miquelduranfrigola** [comment]
Thanks for the updates, @simrantan!

**Comment by simrantan** [comment]
- [ ] Investigate model test error in eos2rd8
- [ ] Test eos3ev6 on CLI, Colab, Docker
- [ ] Test eos3ae6 on CLI, Colab, Docker
- [ ] Try fetching eos78ao on CLI to help Riley 

I tested[ eos3ev6](https://github.com/ersilia-os/eos3ev6/issues/2#issuecomment-1629908196) and [eos3ae6](https://github.com/ersilia-os/eos3ae6/issues/6#issuecomment-1629849662) on all three platforms, and they both ran successfully on eml_canonical. I also helped Riley investigate an issue by fetching a model on my laptop to see if the error he was receiving was a Mac-based one. We found that it persists on my OS, so it is in the model itself. I also thoroughly looked into a new model test error that occurred in actions when eos2rd8 was merged - partway through looking, I saw Riley message about the same error, and communicated with him about it, finding that I was looking in the wrong place for the error source. I then examined the error in more detail, did some research online, and wrote up my findings in[ issues. ](https://github.com/ersilia-os/eos2rd8/issues/2#issuecomment-1629734921).

**Comment by simrantan** [comment]
- [ ] Ersilia Lab Meeting
- [ ] Internship Meeting
- [ ] Set up meeting with @miquelduranfrigola to work on eos2rd8
- [ ] Add a main.py to eos5axz
- [ ] move functionality from service.py to main and convert service to eos-template format in eos5axz
- [ ] test eos5axz locally
- [ ] refactor eos8fth 
- [ ] test eos8fth locally
- [ ] submit PR for eos8fth
- [ ] test eos76wn on CLI, Colab, and Docker 

I began working on eos8fth but found there was a fetch error that led me to discover the model might not be[ ready](https://github.com/ersilia-os/eos8fth/issues/1#issuecomment-1631843453) and I wanted to check on that before refactoring further. I also worked on refactoring eos5axz, which involved adding a main.py, moving functionality from service to main, and re-doing service.py. I spent a while making sure I was getting the format right and not messing up the functionality, but I am getting an error described [here](https://github.com/ersilia-os/eos5axz/issues/6#issuecomment-1631852693) that I could advice on. I also tested eos76wn on CLI, Colab, and Docker and found it worked on CLI and Docker but had a fetch error in [Colab](https://github.com/ersilia-os/eos7w6n/issues/7#issuecomment-1631885097).

**Comment by GemmaTuron** [comment]
Hi @simrantan !


A few updates:
`eos8fth` is indeed not completed, but it would be great if you take it up and try to incorporate it from scratch, it is an Anti Sars Cov2 model
Let's wait for @miquelduranfrigola feedback on `eos5axz` to which extent we want to refactor
And try to debug the cache issue in the actions of eos2rd8.
I think there is quite a lot in your plate already so I won't assign new models for the moment!

**Comment by simrantan** [comment]
- [ ] Look into eos8fth's original issue
- [ ] understand the current progress on eos8fth
- [ ] find the files that need to be completed for eos8fth to run
- [ ] Look into the cache issues of eos2rd8
- [ ] Debug cache issues 
- [ ] make Internship Update presentation

**Comment by simrantan** [comment]
- [ ] Lab Meeting 
- [ ] Internship Meeting
- [ ] Eos5axz - edit service.py to remove checkpoint tracking 
- [ ] Eos8fth - install locally and identify necessary files 
- [ ] Eos8fth - transfer files to eos repo

**Comment by GemmaTuron** [comment]
Thanks for the update @simrantan 

Let's focus on completing the eos8fth, I've asked you a few questions about it in the issue

**Comment by simrantan** [comment]
- [ ] Test eos238c on CLI, Colab, and Docker 
- [ ] Collect information for eos8fth 
- [ ] Create fork of eos8fth and add files from redial-2020
- [ ] Edit paths in code to reflect new folders 
- [ ] sketch out code for input and output adapters from original source code format to ersilia format for eos8fth 

I worked on eos8fth today, collecting information about the two different run_scripts and starting to incorporate the model by moving over the necessary files (including the model pkls). I have also changed the paths to reflect the new folders that information is found in from the original source [code. ](https://github.com/ersilia-os/eos8fth/issues/1#issuecomment-1636493190). I am working on an input adapter, and sketched out some code for what an output adapter could look like. I also tested eos238c on CLI, Colab, and[ Docker. ](https://github.com/ersilia-os/eos238c/issues/3#issuecomment-1636511653)

**Comment by simrantan** [comment]
- [ ] Meeting with Miquel and Zakia about eos2rd8
- [ ] Clean up github - remove codespaces, forks, etc.
- [ ] Implement input and output adapters for eos8fth
- [ ] Test eos1af5 on CLI, Colab, and Docker 
- [ ] Make sure eos5axz works on run, not predict and submit PR

I had a meeting with Miquel and Zakia where they started helping with getting eos2rd8 set up locally so that I could build the image on my device and upload it to work around the actions size issues. I also tested eos1af5 on all [platforms.](https://github.com/ersilia-os/eos1af5/issues/9#issuecomment-1639118824) I took some time to clean up forks of models I am no longer working on and removing codespaces I am not using. I started fresh on eos5axz, removing legacy files and making sure it works with "run", then submitting a [PR. ](https://github.com/ersilia-os/eos5axz/pull/8#issue-1808867873) Finally, I worked on writing an input and output adapter for [eos8fth. ](https://github.com/ersilia-os/eos8fth/issues/1#issuecomment-1639178151)

**Comment by simrantan** [comment]
- [ ] Test eos6tg8
- [ ] Finish output adapter for eos8fth 
- [ ] Finish main.py and run.sh for eos8fth
- [ ] Test eos8fth 
- [ ] Continue fixing dependencies on local version of eos2rd8 on docker 

I tested eos6tg8 on CLI, Colab, and[ Docker ](https://github.com/ersilia-os/eos6tg8/issues/1#issuecomment-1641009938) and it worked great on all platforms. I finished writing the[ output ](https://github.com/ersilia-os/eos8fth/issues/1#issuecomment-1641081459)adapter for eos8fth, and also wrapped up the last steps for incorporating the model but I ran into some [bugs](https://github.com/ersilia-os/eos8fth/issues/1#issuecomment-1641208401) I am working on resolving (and will continue working on tomorrow). I have finished most of the steps for eos8fth, and once I get through these last bugs I should be able to finish testing it and submit it.

**Comment by simrantan** [comment]
- [ ] Investigate eos5axz testing issue 
- [ ] Debug eos8fth
- [ ] Helped Riley look into module import error on eos5smc

Zakia reported an issue with eos5axz, which I looked into today and reported my findings in the[ issues thread](https://github.com/ersilia-os/eos5axz/issues/5#issuecomment-1643076070) as I wait for more details from Zakia. I also spent most of my time debugging eos8fth. I am working through some bugs but have gotten stuck on some that I have[ written abou](https://github.com/ersilia-os/eos8fth/issues/1#issuecomment-1643136152)t. Additionally, I spent some time testing a model for Riley since there was a potential Mac issue, then we worked on looking into possible solutions together that we can test further.

**Comment by simrantan** [comment]
- [ ] Test eos1579 on CLI, Colab, Docker 
- [ ] Debug eos8fth 
- [ ] Look into workflow error for eos8451
- [ ] Refactor eos8451
- [ ] Lab Meeting

I tested [eos1579 ](https://github.com/ersilia-os/eos1579/issues/3#issuecomment-1644763864) on CLI, Colab, and Docker. I continued working on debugging eos8fth, working on the getting the remaining files into lfs.[ Zakia ](https://ersilia-workspace.slack.com/archives/C038FBH00F7/p1689880239477499) helped me look into the bug I was stuck on for eos8fth with the File error, and while I waited for her to try it on her CLI, I also started refactoring eos8451. I found a workflow error that I [wrote up,](https://github.com/ersilia-os/eos8451/issues/1#issuecomment-1644720860) and worked on debugging it. I fixed the issue, and I have refactored the model, but I have a [question ](https://github.com/ersilia-os/eos8451/issues/1#issuecomment-1644744415) about the placement of main.py that I'd like clarification on before I consider it fully finished! Finally, I spent more time debugging eos8fth. With Zakias help, I have fixed the missing File bug, but I am still [working](https://github.com/ersilia-os/eos8fth/issues/1#issuecomment-1644862693) on lfs files and a new error that has occurred. I plan to dig into eos8fth again tomorrow and hopefully resolve this issue.

**Comment by simrantan** [comment]
- [ ] Finish Refactoring eos8451
- [ ] Debug eos8fth
- [ ] Clean up conda environments, forks, cloned repos 

I spent most of today working on bug fixing for [eos8451](https://github.com/ersilia-os/eos8451/issues/1#issuecomment-1646441559) and eos8fth. I managed to find a solution for my previous bug in eos8fth, and am working on testing to find the root of the error in eos8451. I unfortunately ran into a setback when I faced some laptop issues today (my computer shut down unexpectedly and would not turn back on), but managed to resolve it after calling the microsoft help desk. After the shutdown, I was advised to clean up storage, so I decided to do that for Ersilia as well, cleaning up any unnecessary resources.

**Comment by GemmaTuron** [comment]
Hi @simrantan 

Please update the `eos8451` and `eos8fth` before I assign you new tasks. Also, could you update us on the status of building docker images in docker to bypass the git actions size limit?

**Comment by simrantan** [comment]
Hi, 

Sorry! I forgot to hit comment on my internship update for today. Also apologies for the lack of update on building the docker images - I have been working on creating an environment in docker but I haven't had success so I haven't had much progress to update on besides finding different missing libraries. I ran into some issues and wanted to possibly talk to Miquel and Zakia again to work through these problems, but since last week was busy I ended up focusing on eos8fth and other models in the meantime. 

- [ ] Debug eos8fth
- [ ] Debug eos8451

I spent some time working on eos8fth. Fixing the dependencies worked and I am able to run the model. I had to get rid of their many print statements throughout the code that were printing to terminal, so I spent some time hunting those down and removing them. However, I am still having issues with lfs and tracking files on that model, which I have been working on for a long time. I am hoping to resolve Tuesday. but may need some input from anyone who has dealt with similar issues. I also am still working on trying to solve the path issue for eos8451 (the model takes a very long time to fetch, I have to wait a while to see the results of any new path I try). I found Riley is dealing with similar issues and we will tackle it together tomorrow.

**Comment by simrantan** [comment]
- [ ] Finish debugging eos8fth
- [ ] Debug eos8451 
- [ ] Lab Meeting
- [ ] Internship Meeting
- [ ] Begin writeup of Docker Local Image installation

I have made a lot of progress on [eos8fth ](https://github.com/ersilia-os/eos8fth/issues/1#issuecomment-1650868321)and will be able to do final tests and wrap it up tonight or tomorrow depending on how long it takes to review file sizes for the lfs issue which I am still working on. I have also been working on debugging eos8451, and I have been using Febie's model eos85a3 as a guide since it is also a grover model. I am currently testing it, it just takes some time to fetch and will update with the success when it completes. I set up a meeting time with Miquel and Zakia to work on the local docker installation and started recording the process of commands I have been using to try and set up the environment (though it is not working, I am writing it down so that I can either refer back to these commands as a debugging tool or to provide alternative options)

**Comment by simrantan** [comment]
@GemmaTuron @miquelduranfrigola 
- [ ] Test eos8fth and submit PR
- [ ] Test eos8451 and submit PR

As we discussed in the meeting, I am working on wrapping the models I have been working on and will be done with them soon! I would love to take on new models/testing soon so I can continue contributing. 

(pre-meeting update) I have almost finished [eos8fth](https://github.com/ersilia-os/eos8fth/issues/1#issuecomment-1651345983), but there is a [dependency issue](https://github.com/ersilia-os/eos8fth/issues/1#issuecomment-1653060604) that I am unsure how to tackle since it deals with the version of docker ersilia uses and I'm unsure what should or shouldn't be changed for it to work. I also have fixed my prior [issue](https://github.com/ersilia-os/eos8451/issues/1#issuecomment-1653116551) with eos8451

**Comment by simrantan** [comment]
- [ ] Fix eos8fth dependency issue 
- [ ] Fix eos8451 fetch issue

I have been working on both of these issues today and reached points where I would love some outside input if available since I am dealing with things outside of the scope of the models themselves and do not want to mess up the ersilia model hub in any way. [Eos8451](https://github.com/ersilia-os/eos8451/issues/1#issuecomment-1653853466) is dealing with an input issue, and[ eos8fth](https://github.com/ersilia-os/eos8fth/issues/1#issuecomment-1655121137)  is having an empty output error (that is likely a package issue, since this module produces output just fine in a seperate conda environment using bash)

**Comment by simrantan** [comment]
- [ ] Re-start eos8451
- [ ] Test eos8451 
- [ ] Submit PR for eos8451
- [ ] Fix eos8fth 
- [ ] Finish draft of manual upload to dockerhub documentation 
- [ ] Help Riley fetch and run eos5smc 

Last friday, I met with Miquel and discussed the dockerhub issue. We went through the process, and found that the upload itself can only be completed by specific members of the Ersilia organization due to the permissions needed. I wrote out the steps we took to upload the file to Dockerhub and simplified to a general process that can be followed when encountering models that are too large for github actions to upload. I completed the draft of the writeup, and plan to add pictures and more details tomorrow. I also fixed eos8451 and submitted a PR. Inbetween, I helped Riley fetch and test eos5smc. Then, I worked on fixing eos8fth.

**Comment by miquelduranfrigola** [comment]
A comment on `eos2rd8`: This model is big and we need to build it locally. The script to build Docker images locally does not work at the moment, and it does not provide all the funcionality we have in the GitHub Action (https://github.com/ersilia-os/ersilia/blob/master/.github/scripts/build_model_container_and_update_to_dockerhub.py). I am opening a bug report: https://github.com/ersilia-os/ersilia/issues/760

**Comment by simrantan** [comment]
- [ ] Lab Meeting
- [ ] Internship Meeting
- [ ] Make Presentation for Wednesday Meeting
- [ ] Look for correct rdkit version for eos8fth 
- [ ] Test eos8a5g on platforms

Worked on making the presentation for Wednesday's meeting. Tested eos8a5g on all platforms. Messaged Zakia and Hellen about rdkit, and tried looking for versions online. Have been testing different rdkit versions throughout the day (the model takes a while to fetch, so i have to wait some time after trying each version.

**Comment by simrantan** [comment]
8/2 and 8/3
- [ ] Lab Meeting 
- [ ] Internship Meeting 
- [ ] Test eos6o0z
- [ ] Test eos5smc 
- [ ] Test eos96ia
- [ ] Test eos7jlv
- [ ] Specify rdkit descriptors to make newer versions compatible with eos8fth 
- [ ] Fix eos8451 issue 
- [ ] Clean up and Dockerize eos1vms

**Comment by simrantan** [comment]
8/4-8/6

- [ ] Clean and Dockerize[ eos1vms](https://github.com/ersilia-os/eos1vms/pull/3#issue-1837367402)
- [ ] Find source of workflow issue in eos6afz
- [ ] Clean and Dockerize [eos6afz ](https://github.com/ersilia-os/eos6fza/pull/2#issue-1838686765)
- [ ] Check the lens of Descriptors in different rdkit versions to see which has the same number of descriptors as eos8fth requires

I cleaned and refactored the assigned models (and also fixed the workflow error in eos6afz, which turned out to be an rdkit version issue). I have also been downloading different versions of rdkit and checking the number of descriptors. I have checked 5 versions but haven't found an exact match yet and am planning to test more tomorrow so I can get on to the next steps of checking if the descriptors are a match.

**Comment by GemmaTuron** [comment]
Hi @simrantan 

IS model eos8fth ready? Please remember to move the tasks to In progress when you pick up new models.
Regarding the rdkit descriptors, the safest is to:
- Find out the exact list of descriptors used (in the original model) - there are names for them
- Establish those in the newer version you are using, so that if more descriptors are calculated only those that were originally used will be taken into account.

You can get inspiration from this piece of code  under the section #RDKIT Descriptors. HEre, we are selecting 200 specific descriptors
https://github.com/ersilia-os/lazy-qsar/blob/main/lazyqsar/descriptors/descriptors.py

**Comment by simrantan** [comment]
Sorry! I will make sure to move the tasks to in progress in the future. Thank you for the files! I was trying Zakia's method of finding a version of rdkit that does work but it appears it will not be possible so I wrote some code to get the descriptors from the version the source code uses and have written a descriptors.py file based on the example provided! I had a question about how/where to use descriptors.py for it to prevent the issue from happening? Should I use it in the place of importing rdkit?

Also, I moved the finetune folder to checkpoints and re-tested eos6fza and [resubmitted a pr! ](https://github.com/ersilia-os/eos6fza/pull/2#issuecomment-1669037593)

**Comment by GemmaTuron** [comment]
For the descriptors, let's talk about this in today's meeting, I'll guide you through it!
I'll merge the Pr

**Comment by simrantan** [comment]
8/8-8/9

- [ ] Add choosing the 200 specific descriptors needed for eos8fth to get_features.py 
- [ ] Find a version of rdkit that has all 200 specific descriptors *
- [ ] *(new task) write a script to compare the original descriptors from eos8fth source code to descriptors in other versions of rdkit 
- [ ] Test on different rdkit versions until a match is found 
- [ ] Fix eos6fza 
- [ ] Fix eos1vms 
- [ ] Read source code for eos8fma

**Comment by simrantan** [comment]
@GemmaTuron 

I had a flight scheduled for tonight that got cancelled last minute and I was moved to the next available one, which conflicts with tomorrows meeting so I will be unable to attend - I'll look at the presentations in the folder to catch up on the recent updates!

**Comment by GemmaTuron** [comment]
Thanks for letting me know, safe flights.

**Comment by GemmaTuron** [comment]
Hi @simrantan 

Let's focus on closng the open models eos1vms and eos8fth this week, thanks!

**Comment by GemmaTuron** [comment]
Hi @simrantan !

It was great to work with you, thanks so much for your contributions to Ersilia, we hoped you learnt and enjoyed as much as we did! Please, remain engaged with the community and feel free to open any issues or contribute to open ones :)
I'll now close this issue as completed !
